src/main/java/im/status/keycard/KeycardApplet.java

```diff
@@ -1332,6 +1399,31 @@ private void setPinlessPath(APDU apdu) {
 
     JCSystem.commitTransaction();
   }
+  
+  /**
+   * Processes the EXPORT SEED command. Requires an open secure channel and the PIN to be verified.
+   *
+   * @param apdu the JCRE-owned APDU object.
+   */
+  private void exportSeed(APDU apdu) {
+    byte[] apduBuffer = apdu.getBuffer();
+    secureChannel.preprocessAPDU(apduBuffer);
+
+    if (!pin.isValidated() || masterSeedStatus == MASTERSEED_EMPTY) {
+      ISOException.throwIt(ISO7816.SW_CONDITIONS_NOT_SATISFIED);
+    }
+
+    if (masterSeedStatus != MASTERSEED_EXPORTABLE) {
+      ISOException.throwIt(ISO7816.SW_COMMAND_NOT_ALLOWED);
+    }
+
+    short off = SecureChannel.SC_OUT_OFFSET;
+    apduBuffer[off++] = TLV_SEED;
+    apduBuffer[off++] = (byte) BIP39_SEED_SIZE;
+    Util.arrayCopyNonAtomic(masterSeed, (short) 0, apduBuffer, off++, BIP39_SEED_SIZE);
```

Comment:
Due to the secure nature of this data/seed, I recommend you use an atomic copy here. In cases where a power interruption occurs, the apduBuffer may retain the data of the previous copy - it is non-volatile.

See the relevant research paper at the top result of this search:  https://www.google.com/search?client=firefox-b-1-d&q=Transactions+and+non-atomic+API+methods+in+Java+Card

Reply:
Agree with Aurash. Please use `arrayCopy` and wrap in `JCSystem.beginTransaction()`

- [ ] Use atomic `arrayCopy` for `exportSeed`

Reply:
Ahhh good catch. I will change this

Reply:
Actually, I disagree. Atomic copy gains us nothing here. So what if the card is removed before the APDU buffer copy is complete? Nothing bad happens. I can add it, but it would be more code that gains us nothing

Reply:
Per our conversation, I am only going to update ones that save sensitive data in persistent memory